xgboost.py

```diff
@@ -0,0 +1,59 @@
+import sys
+sys.path.append("..")
+
+import pandas as pd
+import numpy as np
+from xgboost import XGBClassifier
+from sklearn.model_selection import StratifiedKFold,cross_val_score
+from sklearn.metrics import f1_score,classification_report,confusion_matrix
+
+from data_preprocessing.preprocessor import Preprocessor
+
+class FraudDetector:
+    def __init__(self,df):
+        self.df = df.copy()
+        self.X_nan = None
+        self.y_nan = None
+        self.df_dropped = None
+        self.X,self.y = None,None
+        self.X_train,self.X_test,self.y_train,self.y_test = None,None,None,None
+        self.xgb = None
+        self.skf = StratifiedKFold(n_splits=5,random_state=42,shuffle=True)
+        
+    def preprocess(self):
+        preprocessor = Preprocessor(self.df)
+        preprocessor.clean()
+        
+    def extract_nan_features(self):
+        self.X_nan = self.df[self.df.isnull().T.any()].drop('FLAG',axis=1)
+        self.y_nan = self.df[self.df.isnull().T.any()]['FLAG']
+        
+    def define_features_labels(self):
+        self.df_dropped = self.df.dropna()
+        self.X,self.y = self.df_dropped.drop('FLAG',axis=1),self.df_dropped['FLAG']
+        
+    def split_data(self):
+        for train_index,test_index in self.skf.split(self.X,self.y):
+            self.X_train,self.X_test = self.X.iloc[train_index],self.X.iloc[test_index],
+            self.y_train,self.y_test = self.y.iloc[train_index],self.y.iloc[test_index]
+            
+    def combine_features(self):
+        y_train_combined = np.concatenate([self.y_train,self.y_nan])
+        X_train_combined = pd.concat([self.X_train,self.X_nan])
+        return X_train_combined, y_train_combined
```

Comment:
<img src="https://avatars.githubusercontent.com/u/116506391?s=400&u=09ae00df628c1f06039a246d04cccb2b4042ed0b&v=4" alt="Image description" width="20" height="20">   CodeSweep

In the `combine_features` method, you are concatenating `self.y_train` and `self.y_nan` without checking if they have the same labels or not. This could potentially introduce inconsistencies in your training data. Ensure that the labels match before combining them.



<!-- This is an auto-generated comment by OSS CodeSweep -->